test/e2e/petset.go

```diff
@@ -223,8 +223,8 @@ var _ = framework.KubeDescribe("PetSet [Slow] [Feature:PetSet]", func() {
 			}
 
 			By("Reading value under foo from member with index 2")
-			if v := pet.read(2, "foo"); v != "bar" {
-				framework.Failf("Read unexpected value %v, expected bar under key foo", v)
+			if err := pollReadWithTimeout(pet, 2, "foo", "bar"); err != nil {
```

Comment:
this error is useless, it's the infamous "timed out waiting for condition". Maybe pipe the actual error reported from db or something? Or say timed out trying to read back value foo from replicated mysql slave...